Sources/UIScrollView+indicatorsInternal.swift

```diff
@@ -6,26 +6,51 @@
 //  Copyright © 2018 flowkey. All rights reserved.
 //
 
-// NOTE: Everything in this file must be private / internal-only.
+// NOTE: Everything in this file must be private or internal.
 // You can't override methods that were defined in an extension.
 
-internal extension UIScrollView {
-    internal static let indicatorDistanceFromScrollViewFrame: CGFloat = 2.5
+extension UIScrollView {
+    private var indicatorLengths: (horizontal: CGFloat, vertical: CGFloat) {
+        let minIndicatorLength: CGFloat = 30.0
+        return (
+            horizontal: max(minIndicatorLength, (bounds.width / contentSize.width) * bounds.width),
+            vertical: max(minIndicatorLength, (bounds.height / contentSize.height) * bounds.height)
+        )
+    }
+
+    private var indicatorOffsetsInContentSpace: (horizontal: CGFloat, vertical: CGFloat) {
+        let indicatorDistanceFromScrollViewFrame: CGFloat = 2.5
 
-    internal func indicatorOffsetsInContentSpace() -> (horizontal: CGFloat, vertical: CGFloat) {
         let totalContentArea = (
             horizontal: contentInset.left + contentSize.width + contentInset.right,
             vertical: contentInset.top + contentSize.height + contentInset.bottom
         )
 
         let scrollViewProgress = (
-            horizontal: (contentInset.left + contentOffset.x) / (totalContentArea.horizontal + bounds.width),
-            vertical: (contentInset.top + contentOffset.y) / (totalContentArea.vertical + bounds.height)
+            horizontal: (contentInset.left + contentOffset.x) / (totalContentArea.horizontal - bounds.width),
+            vertical: (contentInset.top + contentOffset.y) / (totalContentArea.vertical - bounds.height)
+        )
+
+        let shouldShowBothIndicators = shouldLayoutHorizontalScrollIndicator && shouldLayoutVerticalScrollIndicator
+
+        let additionalSpacingToPreventOverlap = (
```

Comment:
Did we make these distances up or are they iOS style? We should comment which

Reply:
done